pkg/controller/dns/consts_for_test.go

```diff
@@ -112,7 +114,8 @@ var happyPathPublicDeployment = &appsv1.Deployment{
 		Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "happy-path-public-external-dns"}},
 		Template: corev1.PodTemplateSpec{
 			ObjectMeta: metav1.ObjectMeta{
-				Labels: map[string]string{"app.kubernetes.io/managed-by": "aks-app-routing-operator", "kubernetes.azure.com/managedby": "aks", "app": "happy-path-public-external-dns", "checksum/configmap": hex.EncodeToString(happyPathPublicJSONHash[:])[:16]},
+				Labels:      map[string]string{"app.kubernetes.io/managed-by": "aks-app-routing-operator", "kubernetes.azure.com/managedby": "aks", "app": "happy-path-public-external-dns", "checksum/configmap": hex.EncodeToString(happyPathPublicJSONHash[:])[:16]},
+				Annotations: map[string]string{"kubernetes.azure.com/set-kube-service-host-fqdn": "true"},
```

Comment:
only additions here is adding the annotations to the testing constants.

These tests should be improved. Manifests package is what's responsible for determining exact resources provisioned by app routing. When we make changes to manifests package it should be tested in the manifests package and we shouldn't need to change things here.

I'll refactor this in a followup PR. Nginx manifests export a type that allows you to get definition for specific resources (like Deployments, configmap, etc). We'll add this improvement to external dns manifests export too and consume them here rather than redefine what should be completely handled by manifest package.